src/QMCDrivers/DMC/SODMCUpdatePbyPFast.cpp

```diff
@@ -85,13 +85,11 @@ void SODMCUpdatePbyPWithRejectionFast::advanceWalker(Walker_t& thisWalker, bool
       ds += std::sqrt(tauovermass/spinMass)*deltaS[iat];
       RealType rr = tauovermass * dot(deltaR[iat], deltaR[iat]);
       rr_proposed += rr;
-      if (rr > m_r2max)
+      if (!W.makeMoveAndCheckWithSpin(iat, dr, ds) || rr > m_r2max)
```

Comment:
No function calls like this in if statements please. The execution order is not obvious. Split.